rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java

```diff
@@ -166,4 +186,63 @@ private Block getByJsonBlockId(String id) {
             }
         }
     }
+
+    private List<TransactionTrace> buildBlockTraces(Block block, TraceFilterRequest traceFilterRequest) {
+        List<TransactionTrace> blockTraces = new ArrayList<>();
+
+        if (block.getNumber() != 0) {
```

Comment:
I think we should "early exit" if `block.getNumber() == 0`, in which case we are not filling `blockTraces` so the branch `if traceFilterRequest != null` is useless, no?